src/d/actor/d_a_btd.cpp

```diff
@@ -1,208 +1,2550 @@
+                if (i_this->m575C[i].ChkTgHit()) {
+                    i_this->m575C[i].OnTgNoConHit();
+                    i_this->m5FEA[i] = 10;
+                    def_se_set_p(i_this, &i_this->m5E98[i + 0x13], i_this->m575C[i].GetTgHitObj(), 0x42);
+                    return;
+                }
+            } else {
+                i_this->m5FEA[i]--;
+            }
+        }
+    }
 }
 
 /* 00001898-000019D8       .text sibuki_set__FP9btd_class */
-void sibuki_set(btd_class*) {
-    /* Nonmatching */
+void sibuki_set(btd_class* i_this) {
+    cXyz local_38;
+
+    btd_sibuki_s* pbVar2 = &i_this->mBtdSibukiS[0];
+    i_this->m6E14 = 0;
+    local_38.x = 0.0f;
+    for (s32 i = 0; i < (s32)ARRAY_SIZE(i_this->mBtdSibukiS); i++, pbVar2++) {
+        if (pbVar2->m00 == 0) {
+            pbVar2->m00 = 1;
+            mDoMtx_YrotS(*calc_mtx, cM_rndFX(32768.0f));
+            local_38.z = REG0_F(13) + (cM_rndFX(200.0f) + 300.0f);
+            local_38.y = cM_rndF(50.0f);
+            MtxPosition(&local_38, &pbVar2->m04);
+            local_38.z = l_HIO.m48 + cM_rndF(l_HIO.m48 * 0.5f);
+            local_38.y = l_HIO.m4C + cM_rndF(l_HIO.m4C * 0.5f);
+            MtxPosition(&local_38, &pbVar2->m10);
+            mDoMtx_YrotS(*calc_mtx, cM_rndFX(32768.0f));
+        }
+    }
 }
 
 /* 000019D8-00001A84       .text kubi_calc__FP9btd_class */
-void kubi_calc(btd_class*) {
-    /* Nonmatching */
+void kubi_calc(btd_class* i_this) {
+    MtxTrans(i_this->m6E50.x, i_this->m6E50.y, i_this->m6E50.z, false);
+    mDoMtx_YrotM(*calc_mtx, i_this->m6E74.y);
+    mDoMtx_XrotM(*calc_mtx, i_this->m6E74.x);
+    mDoMtx_ZrotM(*calc_mtx, i_this->m6E74.z);
+    J3DModel* model = i_this->m02D4->getModel();
+    MTXCopy(*calc_mtx, model->getBaseTRMtx());
+    i_this->m02D4->calc();
+    i_this->m02D8->play();
+    i_this->m02DC->play();
 }
 
 /* 00001A84-000028C4       .text damage__FP9btd_class */
-void damage(btd_class*) {
-    /* Nonmatching */
+s32 damage(btd_class* i_this) {
+    /* Nonmatching - regalloc */
+    static s32 hibi_brk[] = {BTD_BRK_HIBI01, BTD_BRK_HIBI02, BTD_BRK_HIBI03};
+    static u16 hibi_eff_name[] = {dPa_name::ID_SCENE_80C8, dPa_name::ID_SCENE_80C9, dPa_name::ID_SCENE_80CA};
+    static u16 hahen_eff_name[] = {
+        dPa_name::ID_SCENE_80C4,
+        dPa_name::ID_SCENE_80C4,
+        dPa_name::ID_SCENE_80C4,
+        dPa_name::ID_SCENE_80C4,
+        dPa_name::ID_SCENE_80C5,
+        dPa_name::ID_SCENE_80C5,
+        dPa_name::ID_SCENE_80C5,
+        dPa_name::ID_SCENE_80C5,
+        dPa_name::ID_SCENE_80C6,
+        dPa_name::ID_SCENE_80C7,
+        dPa_name::ID_SCENE_80C7,
+        dPa_name::ID_SCENE_80C7
+    };
+    static s32 hahen_eff_index[] = {0x2E, 0x21, 0x24, 0x42, 0x31, 0x27, 0x2A, 0x46, 0x37, 0x20, 0x2D, 0x34};
+    int state;
+    mDoExt_McaMorf* iVar2;
+    J3DAnmTevRegKey* pJVar2;
+    int iVar3;
+    JPABaseEmitter* pJVar4;
+    J3DModel* pJVar8;
+    cXyz local_58;
+
+    iVar2 = (mDoExt_McaMorf*)get_anm(i_this);
```

Comment:
Unnecessary cast. Also applies to the other `get_anm` calls in this TU.
```suggestion
    iVar2 = get_anm(i_this);
```